src/test/opentrackIO_parser/python/opentrackIOlib.py

```diff
@@ -0,0 +1,304 @@
+                if self.verbose:
+                    print("Contents of the parsed sample JSON dict:\n")
+                    print(self.pd)
+                    print("\n")
+        else:
+            print("Parse(): Error. No json text provided!")
+            return -1
+
+    def Conversion_factor_from_meters(self,unit_str):
+        """Stores the multiplicative conversion factor from meters to user-specified units
+        Valid unit specifier strings: m, cm, mm, in"""
+        if unit_str == "m":
+            return 1.0
+        elif unit_str == "cm":
+            return 100.0
+        elif unit_str == "mm":
+            return 1000.0
+        elif unit_str == "in":
+            return (1000/25.4)
+
+    def Get_camera_translation(self, dimension):
+        """Return a single axis of camera translation, such as: x,y, or z"""
+        for tr in self.pd["transforms"]:
+            if ("Camera" in tr["name"]):
+                if self.verbose:
+                    print("found camera, dim = {}, mult factor: {}".format(dimension,self.trans_mult))
+                if (dimension == 'x'):
+                    return tr["translation"]["x"] * self.trans_mult
+                elif (dimension == 'y'):
+                    return tr["translation"]["y"] * self.trans_mult
+                elif (dimension == 'z'):
+                    return tr["translation"]["z"] * self.trans_mult
+                break
+
+    def Get_camera_rotation(self, dimension):
+        """Return a single axis of camera rotation like pan, tilt, or roll.
+        Valid arguments are: p, t, r"""
+        for tr in self.pd["transforms"]:
+            if ("Camera" in tr["name"]):
+                if (dimension == 'p'):
+                    return tr["rotation"]["pan"] * self.rot_mult
+                elif (dimension == 't'):
+                    return tr["rotation"]["tilt"] * self.rot_mult
+                elif (dimension == 'r'):
+                    return tr["rotation"]["roll"] * self.rot_mult
+                break
+
+    def Get_camera_translations(self):
+        """Return 3DOF camera coordinate: (x,y,z)"""
+        return (self.Get_trans('X'), self.Get_trans('Y'), self.Get_trans('Z'))
+
+    def Get_timecode(self):
+        """Return house timecode as a string in LTC format HH:MM:SS:FF"""
+        if ("timing" in self.pd) and (self.pd["timing"]["timecode"]):
+            hh = '{:02}'.format(int(self.pd["timing"]["timecode"]["hours"]))
+            mm = '{:02}'.format(int(self.pd["timing"]["timecode"]["minutes"]))                
+            ss = '{:02}'.format(int(self.pd["timing"]["timecode"]["seconds"]))
+            ff = '{:02}'.format(int(self.pd["timing"]["timecode"]["frames"]))
+            return hh + ':' + mm + ':' + ss + ':' + ff
+        else:
+            return None
+
+    def Get_sample_time(self,part=None):
+        """Get the PTP sample time in user-preferred format"""
+        if ("timing" in self.pd) and (self.pd["timing"]["sampleTimestamp"]):
```

Comment:
You could write a little helper for these tests - something like https://stackoverflow.com/questions/43491287/elegant-way-to-check-if-a-nested-key-exists-in-a-dict

Reply:
Done!